scripts/check-schema-and-codegen/utils.ts

```diff
@@ -0,0 +1,88 @@
+import dns from "dns";
+import fs from "fs";
+import os from "os";
+import { generate } from "@graphql-codegen/cli";
+import { execSync } from "child_process";
+import process from "process";
+import { getConfig } from "../../codegen";
+
+const GITHUB_API = "https://api.github.com";
+const GQL_DIR = "graphql/schema";
+const LOCAL_SCHEMA = "sdlschema";
+const REPO = "/repos/evergreen-ci/evergreen";
+
+/**
+ * Checks if a given domain can be resolved.
+ * @async
+ * @param domain - The domain name to check.
+ * @returns - Resolves to `true` if the domain can be resolved, `false` otherwise.
+ */
+export const canResolveDNS = (domain: string) =>
+  new Promise((resolve) => {
+    dns.lookup(domain, (err) => {
+      if (err) {
+        resolve(false);
+      } else {
+        resolve(true);
+      }
+    });
+  });
+
+/**
+ * Get the latest commit that was made to the GQL folder of the remote Evergreen repository.
+ * @returns A Promise that resolves to the SHA of the latest commit.
+ * @throws {Error} When failed to fetch commits.
+ */
+export async function getLatestCommitFromRemote(): Promise<string> {
+  const url = `${GITHUB_API}${REPO}/commits?path=${GQL_DIR}&sha=main`;
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Failed to fetch ${url}. Status: ${response.status}`);
+  }
+
+  const commits = await response.json();
+
+  if (commits.length > 0) {
+    return commits[0].sha;
+  }
+  throw new Error(`No commits found for this path: ${url}`);
+}
+
+/**
+ * Check if the local Evergreen repo contains a given commit.
+ * @param commit The commit string that will be checked to see if it exists in the Evergreen repo.
+ * @returns A Promise that resolves to true if local repo contains the given commit, and false otherwise.
+ * @throws {Error} When an error occurs while executing the command.
+ */
+export const checkIsAncestor = async (commit: string): Promise<boolean> => {
+  const localSchemaSymlink = fs.readlinkSync(LOCAL_SCHEMA);
+  const originalDir = process.cwd();
+  try {
+    process.chdir(localSchemaSymlink);
+    execSync(`git merge-base --is-ancestor ${commit} HEAD`);
+    process.chdir(originalDir);
+    return true;
+  } catch (error) {
+    process.chdir(originalDir);
+    if (error.status === 1 || error.status === 128) {
```

Comment:
It would be helpful to comment on what these statuses represent or extract them into named constants.